sdk/src/wallet/wallet/builder.rs

```diff
@@ -178,16 +178,27 @@ impl WalletBuilder {
             self.coin_type.replace(coin_type);
         }
 
+        #[cfg(feature = "storage")]
+        let mut accounts = storage_manager.get_accounts().await?;
+        // Check against potential account coin type before saving the wallet data
+        #[cfg(feature = "storage")]
+        if let Some(account) = accounts.first() {
+            // Safe to unwrap, because we made sure that it's Some
```

Comment:
Is this unwrap comment about the expect?

Reply:
Yes

Reply:
Nah that's fine, was confused about something but got it